pkg/fleet/daemon/daemon.go

```diff
@@ -87,7 +87,7 @@ type daemonImpl struct {
 	configs         map[string]installerConfig
 	requests        chan remoteAPIRequest
 	requestsWG      sync.WaitGroup
-	requestsState   map[string]requestState
+	taskDB          *taskDB
```

Comment:
Isn't this https://github.com/DataDog/dd-go/pull/173431? Should we merge that one first?

Reply:
I merged it in this PR since it was require here to test, I can merge the other PR first if you prefer tho

Reply:
no strong opinion, we can keep it in this one and close the other